### PR TITLE
Make `portable` the default in Lighthouse book and Makefile, update workflows

### PR DESCRIPTION
**Pull request by chong-he**
Since `portable` is now the default build, this PR:
- removes some sections in Lighthouse book that are no longer relevant and make `portable` the default where necessary
- updates Makefile to use `portable` as the default build
- removes the `portable` tag in `release.yml` 
- removes the `modern` tag in `docker.yml`

### CODE REVIEW

File: Makefile
Comment (chong-he) [inline Makefile:100]: Not sure if I can delete this and the below `portable` one

Comment (jimmygchen) [reply Makefile:100]: Looks ok to me!

File: book/src/docker.md
Comment (jimmygchen) [inline book/src/docker.md]: modernity is gone from the image tag, so we can remove this, and `modernity` from line 54

**Review by jimmygchen — CHANGES_REQUESTED** [review]
Looks good to me, just added a small comment.
Be good to get a second review from @michaelsproul or @antondlr .

File: book/src/docker.md
Comment (chong-he) [reply book/src/docker.md]: Thank you, removed in [9339a49](https://github.com/sigp/lighthouse/pull/6247/commits/9339a497a4a5616a8fa4a119aede6ae48a339ed1)

**Review by antondlr — APPROVED** [review]
nice, looking good!
now that we're at it I would prefer to remove the `-dev` tag, too. 
no more confusing double tags for any image!

**Review by michaelsproul — APPROVED** [review]
LGTM

Agree we should also get rid of `-dev`

**Review by jimmygchen — APPROVED** [review]
LGTM

**Comment by michaelsproul** [comment]
@mergify queue

**Pull request merged**
